bdb/summarize.c

```diff
@@ -361,15 +368,20 @@ int bdb_summarize_table(bdb_state_type *bdb_state, int ixnum, int comp_pct,
         goto done;
     }
 
-#if defined(_IBM_SOURCE) || defined(__linux__)
+#ifdef POSIX_FADV_SEQUENTIAL
     // inform kernel that we will be accessing file sequentially
-    // and that we won't need the file to be cached
-    fdatasync(fd);
-    posix_fadvise(fd, 0, 0, POSIX_FADV_DONTNEED | POSIX_FADV_SEQUENTIAL);
+    (void)posix_fadvise(fd, 0, 0, POSIX_FADV_SEQUENTIAL);
 #endif
 
     last = comdb2_time_epoch();
-    for (rc = read(fd, page, pgsz); rc == pgsz; rc = read(fd, page, pgsz)) {
+    for (nread = 1, rc = read(fd, page, pgsz); rc == pgsz; ++nread, rc = read(fd, page, pgsz)) {
+#ifdef POSIX_FADV_SEQUENTIAL
+        /* Periodically hint the OS to release pages we've read. Only do so
```

Comment:
How about nread++; here and declare (and init to 0) in the POSIX_FADV_SEQUENTIAL block above.

Reply:
Yes, done.